<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find agv_explore)/map/turtlebot3/turtlebot3_gazebo_world_map.yaml"/>
  <arg name="path_file" default="$(find agv_explore)/map/turtlebot3/turtlebot3_gazebo_world_existing_path.yaml"/>
  <arg name="headless" default="false"/>
  <arg name="move_forward_only" default="false"/>

  <!-- default navigation -->
  <include file="$(find turtlebot3_navigation)/launch/turtlebot3_navigation.launch">
    <arg name="model" value="$(arg model)"/>
    <arg name="map_file" value="$(arg map_file)"/>
    <arg name="open_rviz" value="false"/>
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>
  <param name="/move_base/base_global_planner" value="navfn/ExistingPathGlobalPlanner" />

  <!-- Map server for existing path -->
  <node pkg="map_server" name="map_server_for_existing_path" type="map_server" args="$(arg path_file)">
    <remap from = "/map" to = "/path_map" />
  </node>


  <!-- rviz -->
  <node pkg="rviz" type="rviz" name="rviz"
        args="-d $(find agv_explore)/config/existing_path_global_planner.rviz"
        unless="$(arg headless)"/>


</launch>
